core/app/sorters/spree/products/sort.rb

```diff
@@ -30,6 +31,18 @@ def by_price(scope)
           where(spree_prices: { currency: currency }).
           order("#{Spree::Price.table_name}.amount #{order_direction}")
       end
+
+      def sku?
+        sort_field == 'sku'
+      end
+
+      def by_sku(scope)
+        return scope unless sku?
+
+        scope.joins(:master).
+          where(Spree::Variant.table_name.to_s => { is_master: true }).
+          order("#{Spree::Variant.table_name}.sku #{order_direction}")
```

Comment:
```suggestion
          order(Spree::Variant.table_name => { sku: order_direction })
```
this should work

Reply:
@damianlegawiec Tried that and it doesn't work.